dhl_sdk/db_entities.py

```diff
@@ -0,0 +1,715 @@
+    directly from the database.
+    """
+
+    @abstractmethod
+    def validate_import(self, client: Client) -> bool:
+        """Validate the entity for importing"""
+
+    @abstractmethod
+    def create_request_body(self) -> dict:
+        """Create request body for creating the entity"""
+
+    @staticmethod
+    @abstractmethod
+    def requests(client: Client) -> CRUDClient:
+        """Return a CRUDClient for the entity"""
+
+
+class VariableNumeric(BaseModel):
+    """Model for Numeric Variables"""
+
+    var_type: Literal["numeric"] = "numeric"
+    default: Optional[float] = Field(default=None, alias="default")
+    minimum: Optional[float] = Field(default=None, alias="min")
+    maximum: Optional[float] = Field(default=None, alias="max")
+    interpolation_method: Optional[Literal["linear", "discrete"]] = Field(
+        default=None, alias="interpolation"
+    )
+
+
+class VariableCategorical(BaseModel):
+    """Model for Categorical Variables"""
+
+    var_type: Literal["categorical"] = "categorical"
+    default: Optional[str] = Field(default=None, alias="default")
+    strict: Optional[bool] = Field(default=False, alias="strict")
+    categories: Optional[list[str]] = Field(default=None, alias="values")
+
+
+class VariableLogical(BaseModel):
+    """Model for Logical Variables"""
+
+    var_type: Literal["logical"] = "logical"
+    default: Optional[bool] = Field(alias="default")
+
+
+class FlowVariableReference(BaseModel):
+    """Model for Flow Variable References"""
+
+    measurement_id: str = Field(alias="measurementId")
+    concentration_id: Optional[str] = Field(default=None, alias="concentrationId")
+    fraction_id: Optional[str] = Field(default=None, alias="fractionId")
+
+
+class VariableFlow(BaseModel):
+    """Model for Flow Variables"""
+
+    var_type: Literal["flow"] = "flow"
+    flow_type: Literal["bolus", "conti", "mbolus", "mconti", "sampling", "bleed"] = (
+        Field(default="bolus", alias="type")
+    )
+    step_size: Optional[float] = Field(default=None, alias="stepSize")
+    volume_id: Optional[str] = Field(default=None, alias="volumeId")
+    references: Optional[list[FlowVariableReference]] = Field(
+        default=None, alias="references"
+    )
+
+
+class VariableSpectrum(BaseModel):
+    """Model for Spectrum Variables"""
+
+    var_type: Literal["spectrum"] = "spectrum"
+    default: Optional[list[float]] = Field(alias="default")
+    spectrum: dict = Field(alias="spectrum")
+
+
+VariantDetails = Union[
+    VariableNumeric,
+    VariableCategorical,
+    VariableLogical,
+    VariableFlow,
+    VariableSpectrum,
+]
+
+
+class Group(BaseModel):
+    """Model for Variable Group"""
+
+    id: Optional[str] = Field(default=None, alias="id")
+    code: Optional[str] = Field(default=None, alias="code")
+    name: Optional[str] = Field(default=None, alias="name")
+    aggregation: Optional[str] = Field(default=None, alias="aggregation")
+
```

Comment:
doesnt group also contain processUnits ? but if we dont need em thats ok

Reply:
fyi adam thinks (correctly) that to get a variables group "code" or "name" (i.e. to figure out if a variable is X or Y or whatever) we should not rely on `variable.group.code` (which is smth that can change, its more like a "name"), but we should rely on `dispositions` in any of the matching `processUnit` entities in `variable.group.process_units` (see the recent change i made to our pydantic models in `dhl-io`)